cluster-autoscaler/core/filter_out_schedulable.go

```diff
@@ -63,11 +64,17 @@ func (filterOutSchedulablePodListProcessor) Process(
 	klog.V(4).Infof("Filtering out schedulables")
 	filterOutSchedulableStart := time.Now()
 	var unschedulablePodsToHelp []*apiv1.Pod
+
+	if context.FilterOutSchedulablePodsInUpcomingNodes {
+		unschedulablePodsToHelp = filterOutSchedulableByPacking(unschedulablePods, upcomingNodes, allScheduledPods,
+			context.PredicateChecker, context.ExpendablePodsPriorityCutoff, false)
+	}
+
 	if context.FilterOutSchedulablePodsUsesPacking {
-		unschedulablePodsToHelp = filterOutSchedulableByPacking(unschedulablePods, readyNodes, allScheduledPods,
-			context.PredicateChecker, context.ExpendablePodsPriorityCutoff)
+		unschedulablePodsToHelp = filterOutSchedulableByPacking(unschedulablePodsToHelp, readyNodes, allScheduledPods,
```

Comment:
It can be unset if `FilterOutSchedulablePodsInUpcomingNodes` is false.
The less error prone approach would be to:
* create `tmpUnschedulablePods`
* start with assigning `unschedulablePods` to it.
* then always use the `tmpUnschedulablePods` as both imput and output when `filterOut*` functions are called.
* finally assign `tmpUnschedulablePods` to `unschedulablePodsToHelp`

Reply:
Solved a little differently